File: views/src/api/es-link.js
import request from '@/utils/request'

const api = '/api/es_link/'

export function DeleteAction(data) {
  return request({
    url: api + 'DeleteAction',
    method: 'post',
    data
  })
}

export function ListAction() {
  return request({
    url: api + 'ListAction',
    method: 'get'
  })
}

export function UpdateAction(data) {
  return request({
    url: api + 'UpdateAction',
    method: 'post',
    data
  })
}

export function InsertAction(data) {
  return request({
    url: api + `InsertAction`,
    method: 'post',
    data
  })
}

File: router/es_link.go
package router

import (
	. "github.com/1340691923/ElasticView/controller"

	"github.com/gin-gonic/gin"
)

// ES连接 路由
func runEsLink(app *gin.Engine) {
	esLink := app.Group("/api/es_link")
	{
		esLink.POST("InsertAction", EsLinkController{}.InsertAction)
		esLink.POST("DeleteAction", EsLinkController{}.DeleteAction)
		esLink.POST("UpdateAction", EsLinkController{}.UpdateAction)
		esLink.GET("ListAction", EsLinkController{}.ListAction)
	}
}
